OneSignalSDK/onesignal/src/main/java/com/onesignal/OSNotificationGenerationJob.java

```diff
@@ -0,0 +1,258 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * 1. The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * 2. All copies of substantial portions of the Software may only be used in connection
+ * with services provided by OneSignal.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.onesignal;
+
+
+import android.content.Context;
+import android.net.Uri;
+import android.os.Handler;
+
+import com.onesignal.OneSignal.OSNotificationDisplay;
+
+import org.json.JSONObject;
+
+import java.security.SecureRandom;
+
+public class OSNotificationGenerationJob {
+
+   Context context;
+   JSONObject jsonPayload;
+   boolean isRestoring;
+   boolean isIamPreview;
+   OSNotificationDisplay displayOption = OSNotificationDisplay.NOTIFICATION;
+
+   Long shownTimeStamp;
+
+   CharSequence overriddenBodyFromExtender;
+   CharSequence overriddenTitleFromExtender;
+   Uri overriddenSound;
+   Integer overriddenFlags;
+   Integer orgFlags;
+   Uri orgSound;
+
+   NotificationExtenderService.OverrideSettings overrideSettings;
+
+   OSNotificationGenerationJob(Context context) {
+      this.context = context;
+   }
+
+   CharSequence getTitle() {
+      if (overriddenTitleFromExtender != null)
+         return overriddenTitleFromExtender;
+      return jsonPayload.optString("title", null);
+   }
+   
+   CharSequence getBody() {
+      if (overriddenBodyFromExtender != null)
+         return overriddenBodyFromExtender;
+      return jsonPayload.optString("alert", null);
+   }
+   
+   Integer getAndroidId() {
+      if (overrideSettings == null)
+         overrideSettings = new NotificationExtenderService.OverrideSettings();
+      if (overrideSettings.androidNotificationId == null)
+         overrideSettings.androidNotificationId = new SecureRandom().nextInt();
+      
+      return overrideSettings.androidNotificationId;
+   }
+
+   int getAndroidIdWithoutCreate() {
+      if (overrideSettings == null || overrideSettings.androidNotificationId == null)
+         return -1;
+
+      return overrideSettings.androidNotificationId;
+   }
+
+   /**
+    * If notificationId is -1 then the notification is a silent one
+    */
+   boolean isNotificationToDisplay() {
+      return getAndroidIdWithoutCreate() != -1;
+   }
+
+   String getApiNotificationId() {
+      return OneSignal.getNotificationIdFromFCMJson(jsonPayload);
+   }
+
+   void setAndroidIdWithOutOverriding(Integer id) {
```

Comment:
Without no camelcase

Reply:
maybe setAndroidIdOnOverrideSettings?

Reply:
This is called `setAndroidIdWithoutOverriding` because it only lets you set an Android id in the case where no overrides settings have been set yet and no Android id has been set yet

Reply:
Fixed the camel case issue